test/unit/ts.mjs

```diff
@@ -0,0 +1,55 @@
+import {testPreProcessorExports, batchTest} from '../test-utils.mjs';
+
+import ts from 'typescript';
+
+async function main() {
+  try {
+    let tsOptionsList = [
+      {
+        compilerOptions: {
+          target: 'esnext',
+          module: 'esnext',
+        },
+      },
+      {
+        compilerOptions: {
+          target: 'esnext',
+          module: 'esnext',
+        },
+      },
+      {
+        compilerOptions: {
+          target: ts.ScriptTarget.ESNext,
+          module: ts.ModuleKind.CommonJS,
+        },
+      },
+      {
+        compilerOptions: {
+          target: ts.ScriptTarget.ESNext,
+          module: ts.ModuleKind.ES2015,
+        },
+      },
+    ];
+
+    let preProcessorTypeScript = await import(
+      '../../examples/loaders/preprocessor-typescript.mjs'
+    );
+
+    await testPreProcessorExports(preProcessorTypeScript, tsOptionsList[0]);
+
+    await batchTest(
+      preProcessorTypeScript,
+      tsOptionsList,
+      [
+        'test/fixtures/tsmodule1.ts',
+        'test/fixtures/tsmodule2.ts',
+        'test/fixtures/tsmodule3.ts',
+        'test/fixtures/tsmodule4.ts',
+      ]
+    );
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+main();
```

Comment:
I think this risks rejected promise warnings instead of failed tests. One common pattern is:

```js
tap.test('ts fixtures', async t => {
  await batchTest(t, /* ... */);
});
```

Right now some test failures may just turn into a `console.log` above. Bonus points: When iterating over test cases (in this case: fixture files), it can also be nice to explicitly create subtests (`t.test`) for each. For that you'd likely want to pass the tap interface to `batchTest`, e.g. as the first argument.

Reply:
Ah I see, I'll change to that pattern as well as explicitly calling subtests that way

Reply:
Changed this patter throughout, subtests do make more sense